apps/definition_kafka/lib/kafka/topic/destination.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Kafka.Topic.Destination do
+  use GenServer, shutdown: 30_000
+  use Properties, otp_app: :definition_kafka
+  require Logger
+
+  getter(:dlq, default: Dlq)
+
+  @spec start_link(Destination.t(), Destination.init_opts()) ::
+          {:ok, Destination.t()} | {:error, term}
+  def start_link(topic, opts) do
+    GenServer.start_link(__MODULE__, {topic, opts})
+    |> Ok.map(&%{topic | pid: &1})
+  end
+
+  # TODO logging
+  @spec write(Destination.t(), [term]) :: :ok | {:error, term}
+  def write(topic, [%{} | _] = messages) do
+    encoded =
+      Enum.reduce(messages, %{ok: [], error: []}, fn msg, %{ok: ok, error: err} = acc ->
+        case Jason.encode(msg) do
+          {:ok, json} -> %{acc | ok: [{key(topic, msg), json} | ok]}
+          {:error, reason} -> %{acc | error: [{msg, reason} | err]}
+        end
+      end)
+
+    with :ok <- write(topic, Enum.reverse(encoded.ok)) do
+      GenServer.cast(topic.pid, {:dlq, Enum.reverse(encoded.error)})
+      :ok
+    end
+  end
+
+  def write(topic, messages) do
+    with :ok <- GenServer.call(topic.pid, {:write, topic, messages}),
```

Comment:
if the topic is a process now, why do we need to send it it's own struct definition?

Reply:
I'm going to unmake this a process soon.